Assets/Scripts/WaterCurrentParticles.cs

```diff
@@ -0,0 +1,40 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class WaterCurrentParticles : MonoBehaviour
+{
+    [Tooltip("Emission rate of the particle system.")]
+    [SerializeField] private float emissionRate = 50f;
+    
+    private ParticleSystem particleSystem;
+    private WaterCurrent waterCurrent;
+
+    private void Awake()
+    {
+        waterCurrent = GetComponent<WaterCurrent>();
+        particleSystem = GetComponent<ParticleSystem>();
+    }
+
+    private void Start()
+    {
+        UpdateParticleSystem();
```

Comment:
shouldn't this be called InitializeParticleSystem() if it's only ever called once, at start?